compiler/compile/OMRSymbolReferenceTable.hpp

```diff
@@ -186,15 +186,75 @@ class SymbolReferenceTable
        */
       synchronizedFieldLoadSymbol,
 
-      // common atomic primitives
-      atomicAdd32BitSymbol,
-      atomicAdd64BitSymbol,
+      /** \brief
+       *
+       *  This symbol represents an intrinsic call of the following format:
+       *
+       *  \code
+       *    icall <atomicAddSymbol>
+       *      <address>
+       *      <value>
+       *  \endcode
+       *
+       *  Which performs the following operation atomically:
+       *
+       *  \code
+       *    [address] = [address] + <value>
+       *    return <value>
+       *  \endcode
+       *
+       *  The data type of \c <value> indicates the width of the operation.
+       */
+      atomicAddSymbol,
+
+      /** \brief
+       *
+       *  This symbol represents an intrinsic call of the following format:
+       *
+       *  \code
+       *    icall <atomicFetchAndAddSymbol>
+       *      <address>
+       *      <value>
+       *  \endcode
+       *
+       *  Which performs the following operation atomically:
+       *
+       *  \code
+       *    temp = [address]
+       *    [address] = [address] + <value>
+       *    return temp
+       *  \endcode
+       *
+       *  The data type of \c <value> indicates the width of the operation.
+       */
+      atomicFetchAndAddSymbol,
       atomicFetchAndAdd32BitSymbol,
```

Comment:
I'm assuming these legacy symbol are deprecated, they will be removed once we are satisfied that known downstream projects have adopted this change?

Reply:
That's exactly why they're still here. A subsequent PR will remove them once the downstream dependencies are gone.